pkg/otlp/model/translator/statspayload.go

```diff
@@ -0,0 +1,471 @@
+				}
+			}
+		}
+	}
+	return mmx
+}
+
+// appendSum appends the value val as a sum with the given name to the metric slice. It uses the appropriate fields
+// from tags to set attributes.
+func appendSum(mslice pmetric.MetricSlice, name string, val int64, start, end uint64, tags *pb.ClientGroupedStats) {
+	mx := mslice.AppendEmpty()
+	mx.SetName(name)
+	sum := mx.SetEmptySum()
+	sum.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+	sum.SetIsMonotonic(true)
+
+	dp := sum.DataPoints().AppendEmpty()
+	dp.SetStartTimestamp(pcommon.Timestamp(start))
+	dp.SetTimestamp(pcommon.Timestamp(end))
+	dp.SetIntValue(int64(val))
+	putGroupedStatsAttr(dp.Attributes(), tags)
+}
+
+// appendSketch appends the proto-encoded DDSketch from sketchBytes to the given metric slice as an ExponentialHistogram
+// with the given name, start, and end timestamps. The fields from tags are set as attributes.
+// It is assumed that the DDSketch was created by the trace-agent as a "LogCollapsingLowestDenseDDSketch" with a relative
+// accuracy of 0.001.
+func appendSketch(mslice pmetric.MetricSlice, name string, sketchBytes []byte, start, end uint64, tags *pb.ClientGroupedStats) error {
+	if sketchBytes == nil {
+		// no error, just nothing to do
+		return nil
+	}
+	var msg sketchpb.DDSketch
+	if err := proto.Unmarshal(sketchBytes, &msg); err != nil {
+		return err
+	}
+	dds, err := ddsketch.FromProto(&msg)
+	if err != nil {
+		return err
+	}
+	if dds.IsEmpty() {
+		// no error, just nothing to do
+		return nil
+	}
+	mx := mslice.AppendEmpty()
+	mx.SetName(name)
+	hist := mx.SetEmptyExponentialHistogram()
+	hist.SetAggregationTemporality(pmetric.AggregationTemporalityDelta)
+
+	dp := hist.DataPoints().AppendEmpty()
+	dp.SetStartTimestamp(pcommon.Timestamp(start))
+	dp.SetTimestamp(pcommon.Timestamp(end))
+	putGroupedStatsAttr(dp.Attributes(), tags)
+	dp.SetCount(uint64(dds.GetCount()))
+	if max, err := dds.GetMaxValue(); err == nil {
+		dp.SetMax(max)
+	}
+	if min, err := dds.GetMinValue(); err == nil {
+		dp.SetMin(min)
+	}
+	dp.SetSum(dds.GetSum())
+	dp.SetZeroCount(uint64(dds.GetZeroCount()))
+	// Relative accuracy (ra) in the trace-agent is 0.001. See:
+	// https://github.com/DataDog/datadog-agent/blob/198e1a/pkg/trace/stats/statsraw.go#L19-L21
+	// Gamma is computed as (1+ra)/(1-ra), which results in 1.02020202. Another formula for gamma
+	// is 2^2^-scale. Using that formula, we conclude that a scale of 5 should be suitable, which
+	// is equal to a gamma of 1.0218971486541166.
+	//
+	// It will not be equally accurate, but the error margin is negligible. Generally, the
+	// ExponentialHistogram is simply a data recipient being converted back to the original DDSketch,
+	// at which point we know the original gamma and the resulting sketch will be equivalent. The
+	// only scenario when the histogram could be used is if someone exports it using a non-Datadog
+	// exporter.
```

Comment:
I think this should be fine, but if we want to be fully accurate, we can change the mapping on the ddsketch itself instead of doing this.

Reply:
We want the ddsketch to be identical and consistent with what the trace-agent / exporter generates in all other cases. The ddsketch is the important piece of truth. It must not be changed. The ExponentialHistogram is just an intermediate structure temporarily holding the data. It should never be used. If it is ever used, it won't be with Datadog and in that case a little inaccuracy does not matter.

Reply:
The difference seems negligible so I guess we can ignore this for now and revisit if it becomes a problem